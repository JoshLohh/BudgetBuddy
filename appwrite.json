{
    "projectId": "683b7b97000aae994ff5",
    "projectName": "BudgetBuddy",
    "settings": {
        "services": {
            "account": true,
            "avatars": true,
            "databases": true,
            "locale": true,
            "health": true,
            "storage": true,
            "teams": true,
            "users": true,
            "sites": true,
            "functions": true,
            "graphql": true,
            "messaging": true
        },
        "auth": {
            "methods": {
                "jwt": true,
                "phone": true,
                "invites": true,
                "anonymous": true,
                "email-otp": true,
                "magic-url": true,
                "email-password": true
            },
            "security": {
                "duration": 31536000,
                "limit": 0,
                "sessionsLimit": 10,
                "passwordHistory": 0,
                "passwordDictionary": false,
                "personalDataCheck": false,
                "sessionAlerts": false,
                "mockNumbers": []
            }
        }
    },
    "functions": [
        {
            "$id": "685d1db2003731b3c3d6",
            "execute": [
                "any"
            ],
            "name": "CreateUserProfile",
            "enabled": true,
            "logging": true,
            "runtime": "node-18.0",
            "scopes": [
                "users.read",
                "databases.read",
                "databases.write",
                "collections.read",
                "collections.write",
                "attributes.read",
                "attributes.write",
                "indexes.read",
                "indexes.write",
                "documents.read",
                "documents.write"
            ],
            "events": [
                "users.create"
            ],
            "schedule": "",
            "timeout": 15,
            "entrypoint": "src/main.js",
            "commands": "npm install",
            "specification": "s-0.5vcpu-512mb",
            "path": "functions/CreateUserProfile"
        },
        {
            "$id": "685d26ee003371a58959",
            "execute": [
                "any"
            ],
            "name": "create-user",
            "enabled": true,
            "logging": true,
            "runtime": "bun-1.1",
            "scopes": [
                "users.read",
                "databases.read",
                "databases.write",
                "collections.read",
                "collections.write",
                "attributes.read",
                "attributes.write",
                "indexes.read",
                "indexes.write",
                "documents.read",
                "documents.write"
            ],
            "events": [
                "users.*.create"
            ],
            "schedule": "",
            "timeout": 15,
            "entrypoint": "src/main.ts",
            "commands": "bun install",
            "specification": "s-0.5vcpu-512mb",
            "path": "functions/create-user"
        }
    ],
    "databases": [
        {
            "$id": "684bd404003542c8a2ac",
            "name": "budget_buddy_db",
            "enabled": true
        }
    ],
    "collections": [
        {
            "$id": "684bd42e0009b1e585e7",
            "$permissions": [
                "create(\"users\")",
                "read(\"users\")",
                "update(\"users\")",
                "delete(\"users\")"
            ],
            "databaseId": "684bd404003542c8a2ac",
            "name": "groups",
            "enabled": true,
            "documentSecurity": true,
            "attributes": [
                {
                    "key": "title",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "members",
                    "type": "string",
                    "required": false,
                    "array": true,
                    "size": 500,
                    "default": null
                },
                {
                    "key": "createdBy",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "description",
                    "type": "string",
                    "required": false,
                    "array": false,
                    "size": 500,
                    "default": null
                }
            ],
            "indexes": []
        },
        {
            "$id": "684bd5ef0030ac0cc7f7",
            "$permissions": [],
            "databaseId": "684bd404003542c8a2ac",
            "name": "expenses",
            "enabled": true,
            "documentSecurity": false,
            "attributes": [
                {
                    "key": "groupId",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "title",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "amount",
                    "type": "double",
                    "required": true,
                    "array": false,
                    "min": 0,
                    "max": 1.7976931348623157e+308,
                    "default": null
                },
                {
                    "key": "paidBy",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "splitBetween",
                    "type": "string",
                    "required": false,
                    "array": true,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "splitType",
                    "type": "string",
                    "required": false,
                    "array": false,
                    "size": 255,
                    "default": "equal"
                },
                {
                    "key": "customSplit",
                    "type": "string",
                    "required": false,
                    "array": false,
                    "size": 1000,
                    "default": null
                }
            ],
            "indexes": []
        },
        {
            "$id": "684bd8a5003d02f9b84d",
            "$permissions": [],
            "databaseId": "684bd404003542c8a2ac",
            "name": "settlements",
            "enabled": true,
            "documentSecurity": false,
            "attributes": [
                {
                    "key": "groupId",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "from",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "to",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "amount",
                    "type": "double",
                    "required": true,
                    "array": false,
                    "min": 0,
                    "max": 1.7976931348623157e+308,
                    "default": null
                },
                {
                    "key": "date",
                    "type": "datetime",
                    "required": true,
                    "array": false,
                    "format": "",
                    "default": null
                }
            ],
            "indexes": []
        },
        {
            "$id": "685cf12e003976d23b39",
            "$permissions": [
                "create(\"any\")",
                "read(\"any\")",
                "update(\"any\")",
                "delete(\"any\")"
            ],
            "databaseId": "684bd404003542c8a2ac",
            "name": "users",
            "enabled": true,
            "documentSecurity": true,
            "attributes": [
                {
                    "key": "userId",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "username",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "email",
                    "type": "string",
                    "required": true,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "avatar",
                    "type": "string",
                    "required": false,
                    "array": false,
                    "size": 255,
                    "default": null
                },
                {
                    "key": "bio",
                    "type": "string",
                    "required": false,
                    "array": false,
                    "size": 1000,
                    "default": null
                }
            ],
            "indexes": []
        }
    ]
}