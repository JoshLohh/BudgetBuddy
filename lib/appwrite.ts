import { Client, Account, ID, Avatars } from 'react-native-appwrite';


export const client = new Client()
    .setEndpoint('https://fra.cloud.appwrite.io/v1')
    .setProject("683b7b97000aae994ff5")
    .setPlatform('dev.buddies.BudgetBuddy');

export const account = new Account(client)
export const avatars = new Avatars(client)