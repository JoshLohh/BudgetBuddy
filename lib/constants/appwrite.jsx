import { Client, Account, ID, Avatars } from 'react-native-appwrite';


export const client = new Client
    .setProject("6838ad0d003d2a0754a0")
    .setPlatform('dev.Buddies.budgetbuddy');

export const account = new Account(client)
export const avatars = new Avatars(client)