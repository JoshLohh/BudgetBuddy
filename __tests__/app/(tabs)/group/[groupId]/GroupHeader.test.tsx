/**
 * Test suite for GroupHeader component
 * Covers rendering, editing, avatar upload, backend integration,
 * error cases, and navigation.
 * Mocks Appwrite, fetch, Expo ImagePicker, and Expo Router.
 */

import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import GroupHeader from '@/app/(tabs)/group/[groupId]/GroupHeader';
import { databases } from '@/lib/appwrite';
import { router } from 'expo-router';
import * as ImagePicker from 'expo-image-picker';

// ----------- Mocks -----------
jest.mock('@/lib/appwrite', () => ({
  databases: { updateDocument: jest.fn() },
}));
jest.mock('expo-router', () => ({
  router: { push: jest.fn() }
}));
jest.mock('expo-image-picker', () => ({
  launchImageLibraryAsync: jest.fn(),
  requestMediaLibraryPermissionsAsync: jest.fn(),
}));
jest.mock('@expo/vector-icons', () => ({
    Ionicons: () => null, // Mock to a no-op component
  }));
  jest.mock('expo-image', () => ({
    Image: () => null,
  }));
  jest.mock('expo-font', () => ({
    loadAsync: jest.fn(),
  }));
  jest.mock('expo-asset', () => ({
    fromModule: jest.fn(() => ({ downloadAsync: jest.fn(), localUri: 'mock' })),
  }));

global.fetch = jest.fn().mockResolvedValue({
    ok: true,
    json: async () => ({ $id: 'mockFileId' }),
  }) as unknown as typeof fetch;
  
// ----------- Test data -----------
const fakeGroup = {
  $id: "grp1",
  title: "Trip to Tokyo",
  description: "Spring vacation",
  createdBy: "user1",
  members: ["user1", "user2", "user3"],
  avatar: null,
};

const updatedGroup = {
  ...fakeGroup,
  title: "Tokyo 2025",
  description: "Updated trip!",
};

const avatarUrl = "https://example.com/avatar.jpg";

// ----------- Tests -----------
describe('GroupHeader', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    (databases.updateDocument as jest.Mock).mockResolvedValue({});
    (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: true });
    (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({ canceled: true });
    global.fetch = jest.fn().mockResolvedValue({
        ok: true,
        json: async () => ({ $id: 'mockFileId' }),
      }) as unknown as typeof fetch;
    });
  

  it('renders group info properly', () => {
    const { getByText } = render(
      <GroupHeader group={fakeGroup} totalExpenses={123.75} />
    );
    expect(getByText("Trip to Tokyo")).toBeTruthy();
    expect(getByText("Spring vacation")).toBeTruthy();
    expect(getByText("3 members")).toBeTruthy();
    expect(getByText("Total")).toBeTruthy();
    expect(getByText("$123.75")).toBeTruthy();
  });

  //Snapshots
  it('matches snapshot in view mode', () => {
    const tree = render(<GroupHeader group={fakeGroup} totalExpenses={42.50} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //Snapshot
  it('matches snapshot in edit mode', () => {
    const { getByText, toJSON, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={99.00} />);
    //fireEvent.press(getByText("Edit"));
    fireEvent.press(getByTestId('edit-button'));
    expect(toJSON()).toMatchSnapshot();
  });

  it('enters edit mode and allows saving of new title & description', async () => {
    const onGroupUpdated = jest.fn();
    const { getByTestId, getByDisplayValue, getByText } = render(
      <GroupHeader group={fakeGroup} totalExpenses={79.6} onGroupUpdated={onGroupUpdated} />
    );
    fireEvent.press(getByTestId('edit-button'));
    fireEvent.changeText(getByDisplayValue('Trip to Tokyo'), 'Tokyo 2025');
    fireEvent.changeText(getByDisplayValue('Spring vacation'), 'Updated trip!');
    await act(async () => {
      fireEvent.press(getByText('Save'));
    });
    expect(databases.updateDocument).toHaveBeenCalled();
    expect(onGroupUpdated).toHaveBeenCalledWith(
      expect.objectContaining({ title: 'Tokyo 2025', description: 'Updated trip!' })
    );
  });
  it('shows error when save fails', async () => {
    (databases.updateDocument as jest.Mock).mockRejectedValueOnce(new Error('fail'));
    const { getByText, getByDisplayValue, findByText, getByTestId } = render(
      <GroupHeader group={fakeGroup} totalExpenses={79.60} />
    );
    fireEvent.press(getByTestId('edit-button'));

    await act(async () => {
      fireEvent.press(getByText("Save"));
    });
    expect(await findByText("Failed to update group details.")).toBeTruthy();
  });

  it('calls router.push for activity log and report buttons', () => {
    const { getByText, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={10} />);
    fireEvent.press(getByText("View Activity Log"));
    expect(router.push).toHaveBeenCalledWith({
      pathname: '/group/[groupId]/history',
      params: { groupId: fakeGroup.$id }
    });
    fireEvent.press(getByTestId('report-button'));
    expect(router.push).toHaveBeenCalledWith({
      pathname: '/group/[groupId]/report',
      params: { groupId: fakeGroup.$id }
    });
  });

  it('starts avatar picker and updates avatar on success', async () => {
    (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({
      canceled: false,
      assets: [{ uri: "fakeuri.jpg", fileName: "file.jpg", mimeType: "image/jpeg" }]
    });
    global.fetch = jest.fn().mockResolvedValue({
        ok: true,
        json: async () => ({ $id: 'mockFileId' }),
      }) as unknown as typeof fetch;
    

    const nextAvatarUrl = "https://testendpoint/avatars/mockFileId/view?project=testproject&t=123";
    const onGroupUpdated = jest.fn();
    const { getByLabelText } = render(
      <GroupHeader group={fakeGroup} totalExpenses={12} onGroupUpdated={onGroupUpdated} />
    );
    // Simulate avatar edit button (update selector as needed)
    // fireEvent.press(getByLabelText("Edit Avatar"));
    // For a real test, assign proper testID or accessibilityLabel to avatar edit touchable

    // You can now advance assertions once handleAvatarChange is invoked

    // expect(onGroupUpdated).toHaveBeenCalledWith(expect.objectContaining({ avatar: nextAvatarUrl }));
  });

  it('displays only title and correct members label for 1 member, does not crash with missing desc/avatar', () => {
    const oneMemberGroup = { ...fakeGroup, members: ["user1"], description: undefined, avatar: undefined };
    const { getByText, queryByText } = render(
      <GroupHeader group={oneMemberGroup} totalExpenses={999} />
    );
    expect(getByText("Trip to Tokyo")).toBeTruthy();
    expect(getByText("1 member")).toBeTruthy();
    expect(queryByText("Spring vacation")).toBeFalsy();
  });
});

//Next few tests focus on avatar change handling
describe('handleAvatarChange', () => {
    beforeAll(() => {
        global.alert = jest.fn();
      });

    beforeEach(() => {
      jest.clearAllMocks();
    });
  
    it('alerts and returns if permissions are not granted', async () => {
  (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: false });
  const { getByText, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={0} />);
  //fireEvent.press(getByText('Edit Avatar'));
  fireEvent.press(getByTestId('avatar-edit'));

  await waitFor(() => {
    expect(global.alert).toHaveBeenCalledWith('Permission to access gallery is required!');
  });
});
  
    it('does nothing if user cancels image picking', async () => {
      (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: true });
      (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({ canceled: true });
      
      const { getByText, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={0} />);
      
      fireEvent.press(getByTestId('avatar-edit'));
      
      await new Promise(res => setTimeout(res, 0)); // next tick
      
      expect(fetch).not.toHaveBeenCalled();
    });
  
    it('uploads image and updates group on successful upload', async () => {
      (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: true });
      (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({
        canceled: false,
        assets: [{ uri: 'file://myimage.jpg', mimeType: 'image/jpeg' }]
      });
      global.fetch = jest.fn().mockResolvedValue({
        ok: true,
        json: async () => ({ $id: '123' }),
      }) as unknown as typeof fetch;
    
      (databases.updateDocument as jest.Mock).mockResolvedValue({});
  
      const mockCallback = jest.fn();
      const { getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={0} onGroupUpdated={mockCallback} />);
      
      fireEvent.press(getByTestId('avatar-edit'));
      
      await waitFor(() => {
        expect(fetch).toHaveBeenCalled();
        expect(databases.updateDocument).toHaveBeenCalled();
        expect(mockCallback).toHaveBeenCalledWith(expect.objectContaining({
          avatar: expect.stringContaining('/123')
        }));
      });
    });
  
    it('handles fetch json parse error gracefully', async () => {
      (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: true });
      (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({
        canceled: false,
        assets: [{ uri: 'file://myimage.jpg', mimeType: 'image/jpeg' }]
      });
      const fakeResponse = {
        ok: true,
        json: jest.fn().mockImplementation(() => { throw new Error('invalid json'); })
      };
      global.fetch = jest.fn().mockResolvedValue(fakeResponse) as unknown as typeof fetch;
    
    
      
      const alertMock = jest.spyOn(global, 'alert').mockImplementation(() => {});
  
      const { getByText, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={0} />);
      
      fireEvent.press(getByTestId('avatar-edit'));
      
      await waitFor(() => {
        expect(alertMock).toHaveBeenCalledWith('Upload failed: invalid response from server.');
      });
  
      alertMock.mockRestore();
    });
  
    it('handles fetch non-ok response gracefully', async () => {
      (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: true });
      (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({
        canceled: false,
        assets: [{ uri: 'file://myimage.jpg', mimeType: 'image/jpeg' }]
      });
      global.fetch = jest.fn().mockResolvedValue({
        ok: false,
        json: async () => ({ message: 'Server error' }),
      }) as unknown as typeof fetch;
    
  
      const alertMock = jest.spyOn(global, 'alert').mockImplementation(() => {});
  
      const { getByText, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={0} />);
      
      fireEvent.press(getByTestId('avatar-edit'));
      
      await waitFor(() => {
        expect(alertMock).toHaveBeenCalledWith('Upload failed: Server error');
      });
      alertMock.mockRestore();
    });
  
    it('handles update document failure gracefully', async () => {
      (ImagePicker.requestMediaLibraryPermissionsAsync as jest.Mock).mockResolvedValue({ granted: true });
      (ImagePicker.launchImageLibraryAsync as jest.Mock).mockResolvedValue({
        canceled: false,
        assets: [{ uri: 'file://myimage.jpg', mimeType: 'image/jpeg' }]
      });

      global.fetch = jest.fn().mockResolvedValue({
        ok: true,
        json: async () => ({ $id: '123' }),
      }) as unknown as typeof fetch;
    
      (databases.updateDocument as jest.Mock).mockRejectedValue(new Error('Failed to update'));
  
      const alertMock = jest.spyOn(global, 'alert').mockImplementation(() => {});
      const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
  
      const { getByText, getByTestId } = render(<GroupHeader group={fakeGroup} totalExpenses={0} />);
      
      fireEvent.press(getByTestId('avatar-edit'));
      
      await waitFor(() => {
        expect(alertMock).toHaveBeenCalledWith('Error updating avatar');
        expect(consoleSpy).toHaveBeenCalled();
      });
  
      alertMock.mockRestore();
      consoleSpy.mockRestore();
    });
  });
  