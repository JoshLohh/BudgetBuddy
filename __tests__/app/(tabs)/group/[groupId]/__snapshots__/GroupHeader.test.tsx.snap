// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupHeader matches snapshot in edit mode 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "backgroundColor": "#fff",
        "borderRadius": 12,
        "marginBottom": 10,
        "padding": 10,
      },
    ]
  }
  testID="themed-view"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "marginRight": 12,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontWeight": "bold",
              "marginBottom": 4,
              "marginLeft": 4,
            },
          ]
        }
      >
        Title
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#d6d5e1",
              "borderRadius": 6,
              "color": "#11181C",
              "padding": 20,
            },
            {
              "marginBottom": 8,
            },
          ]
        }
        value="Trip to Tokyo"
      />
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "fontWeight": "bold",
              "marginBottom": 4,
              "marginLeft": 4,
            },
          ]
        }
      >
        Description
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          [
            {
              "backgroundColor": "#d6d5e1",
              "borderRadius": 6,
              "color": "#11181C",
              "padding": 20,
            },
            {
              "marginBottom": 8,
            },
          ]
        }
        value="Spring vacation"
      />
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "minWidth": 110,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#b3e0ff",
            "borderRadius": 10,
            "elevation": 2,
            "marginBottom": 10,
            "minWidth": 110,
            "paddingHorizontal": 18,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#1976d2",
                "fontSize": 13,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Total
        </Text>
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#1976d2",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginTop": 2,
              },
            ]
          }
        >
          $
          99.00
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderRadius": 6,
              "marginVertical": 10,
              "padding": 18,
            },
            {
              "backgroundColor": "#0a7ea4",
            },
            false,
            {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "#0a7ea4",
              "borderRadius": 8,
              "justifyContent": "center",
              "paddingHorizontal": 0,
              "paddingVertical": 10,
              "width": 110,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#fff",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderRadius": 6,
              "marginVertical": 10,
              "padding": 18,
            },
            {
              "backgroundColor": "#0a7ea4",
            },
            false,
            [
              {
                "alignItems": "center",
                "alignSelf": "flex-end",
                "backgroundColor": "#0a7ea4",
                "borderRadius": 8,
                "justifyContent": "center",
                "paddingHorizontal": 0,
                "paddingVertical": 10,
                "width": 110,
              },
              {
                "backgroundColor": "#e0e0e0",
                "marginTop": 8,
              },
            ],
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#333",
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`GroupHeader matches snapshot in view mode 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "backgroundColor": "#fff",
        "borderRadius": 12,
        "marginBottom": 10,
        "padding": 10,
      },
    ]
  }
  testID="themed-view"
>
  <View
    style={
      {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "marginRight": 12,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 12,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "marginBottom": 12,
              "opacity": 1,
            }
          }
          testID="avatar-edit"
        />
        <View
          style={
            {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                undefined,
                {
                  "fontSize": 28,
                  "fontWeight": "bold",
                  "lineHeight": 32,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "black",
                  "fontSize": 28,
                  "fontWeight": "700",
                },
              ]
            }
          >
            Trip to Tokyo
          </Text>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "#555",
                  "fontSize": 15,
                  "marginBottom": 4,
                },
              ]
            }
          >
            Spring vacation
          </Text>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "#888",
              "fontSize": 14,
              "marginBottom": 8,
            },
          ]
        }
      >
        3
         member
        s
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderRadius": 6,
              "marginVertical": 10,
              "padding": 18,
            },
            {
              "backgroundColor": "#0a7ea4",
            },
            false,
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "height": 32,
              "justifyContent": "center",
              "marginBottom": 4,
              "marginTop": 4,
              "minWidth": 200,
              "paddingHorizontal": 14,
              "paddingVertical": 0,
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              {
                "color": "#fff",
                "fontSize": 12,
                "textAlign": "center",
              },
            ]
          }
        >
          View Activity Log
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "flex-end",
          "minWidth": 110,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="report-button"
      >
        <View
          style={
            {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "#b3e0ff",
              "borderRadius": 10,
              "elevation": 2,
              "marginBottom": 10,
              "minWidth": 110,
              "paddingHorizontal": 18,
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "#1976d2",
                  "fontSize": 13,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Total
          </Text>
          <Text
            style={
              [
                {
                  "color": "#11181C",
                },
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                {
                  "color": "#1976d2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginTop": 2,
                },
              ]
            }
          >
            $
            42.50
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderRadius": 6,
              "marginVertical": 10,
              "padding": 18,
            },
            {
              "backgroundColor": "#0a7ea4",
            },
            false,
            {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "#0a7ea4",
              "borderRadius": 8,
              "justifyContent": "center",
              "paddingHorizontal": 0,
              "paddingVertical": 10,
              "width": 110,
            },
          ]
        }
        testID="edit-button"
      />
    </View>
  </View>
</View>
`;
